5.x

t-works opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->
## Subject

<!-- Describe your Pull Request content here -->

I am targeting this branch, because this fixes issues in current branch 5.x and because it is supposed to work && tested with symfony 6.1+.

This is rebased version.

<!--
    Specify which issues will be fixed/closed.
    Remove it if this is not related.
-->

I fixed issues preventing CKEditor upload and browsing 
Closes #669

## Changelog

FIXED: CKEditor upload and browse files with symfony 6.1 and doctrine-orm-admin-bundle

<!-- REMOVE EMPTY SECTIONS -->
```markdown
### Added
- Added service configuration for Sonata\FormatterBundle\Controller\CkeditorAdminController

### Fixed
- browser.html.twig (compatibility with doctrine-orm-admin-bundle)
- upload.html.twig 

## To do
- [ ] Fix label translation in file browser filters;
- [ ] Update the documentation;
- [ ] Allow using custom contexts. (currently using "context:  my-context" in fos_ck_editor.configs.default of tha application does not work with file browser)

t-works committed:
Fix issues preventing CKEditor upload and browsing

t-works committed:
Fix issues preventing CKEditor upload and browsing, rebase.

t-works committed:
Merge remote-tracking branch 'origin/5.x' into 5.x

# Conflicts:
#	src/Resources/views/Ckeditor/browser.html.twig
#	src/Resources/views/Ckeditor/upload.html.twig

t-works committed:
white space

jordisala1991 commented:
Now the bundle tests browser and upload routes, with functional tests using doctrine orm admin bundle, from 4.4 up to 6.1 versions of Symfony.

Can you check the 5.x-dev (without this PR) on your project to see if everything works?

VincentLanglet commented:
@jordisala1991 Now your PR is merged, is there additional fixes that we have to take from this one ?

jordisala1991 commented:
> @jordisala1991 Now your PR is merged, is there additional fixes that we have to take from this one ?

I don't know yet, that's why I have asked @t-works to tests the 5.x-dev branch.

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

t-works commented:
>Now your PR is merged, is there additional fixes that we have to take from this one ?

The browse server action in FCKEditor triggers this errror:

Neither the property "sortparameters" nor one of the methods "sortparameters()", "getsortparameters()"/"issortparameters()"/"hassortparameters()" or "__call()" exist and have public access in class 
"Sonata\DoctrineORMAdminBundle\Model\ModelManager".

Twig\Error\
RuntimeError
in [C:\var\projects\stronaklubu2\vendor\sonata-project\formatter-bundle\src\Resources\views\Ckeditor\browser.html.twig ](https://127.0.0.1:8000/_profiler/open?file=vendor\sonata-project\formatter-bundle\src\Resources\views\Ckeditor\browser.html.twig&line=97#line97)(line 97)

{% set sort_parameters      = datagrid.sortparameters(field_description, datagrid)|merge(ckParameters) %}

I described reason in previous discussion - it's got to be:
 {% set sort_parameters      = datagrid.sortparameters(field_description, datagrid)|merge(ckParameters) %}

The upload action works fine.

BTW please note I had to use a fork of the 
https://github.com/coopTilleuls/CoopTilleulsCKEditorSonataMediaBundle
to use it with symfony6
the fork with dependencies pushed up and changed name to avoid namespace clash is here:
https://github.com/t-works/CKEditorSonataMediaBundleFork

>Could you please rebase your PR and fix merge conflicts?

I fixed merge conflicts, removed xml config since you rewrote it to PHP.

A note on tests:

public function uploadAction(Request $request, MediaManagerInterface $mediaManager): Response
....
media = $mediaManager->create();

2 arguments required here.

t-works committed:
Merge branch '5.x' into 5.x

t-works committed:
Update CkeditorAdminController.php

t-works committed:
Delete ckeditor.xml

VincentLanglet commented:
> > Now your PR is merged, is there additional fixes that we have to take from this one ?
> 
> The browse server action in FCKEditor triggers this errror:
> 
> Neither the property "sortparameters" nor one of the methods "sortparameters()", "getsortparameters()"/"issortparameters()"/"hassortparameters()" or "__call()" exist and have public access in class "Sonata\DoctrineORMAdminBundle\Model\ModelManager".
> 
> Twig\Error RuntimeError in [C:\var\projects\stronaklubu2\vendor\sonata-project\formatter-bundle\src\Resources\views\Ckeditor\browser.html.twig ](https://127.0.0.1:8000/_profiler/open?file=vendor%5Csonata-project%5Cformatter-bundle%5Csrc%5CResources%5Cviews%5CCkeditor%5Cbrowser.html.twig&line=97#line97)(line 97)
> 
> {% set sort_parameters = datagrid.sortparameters(field_description, datagrid)|merge(ckParameters) %}
> 
> I described reason in previous discussion - it's got to be: {% set sort_parameters = datagrid.sortparameters(field_description, datagrid)|merge(ckParameters) %}

Thanks, we will add more functionnals tests and fix this

jordisala1991 commented:
This should not be needed anymore. Closing for now. Have a look at 5.x branch now.

VincentLanglet commented:
This should now be covered by https://github.com/sonata-project/SonataFormatterBundle/pull/693
